substrate/primitives/runtime/src/lib.rs

```diff
@@ -1051,6 +1051,29 @@ pub use alloc::borrow::Cow;
 #[deprecated = "Use String or Cow<'static, str> instead"]
 pub type RuntimeString = alloc::string::String;
 
+/// A target for `core::write!` macro - constructs a string in memory.
+#[derive(Default)]
+pub struct Writer(vec::Vec<u8>);
```

Comment:
We could just replace the usage of writer with `alloc::string::String` to no require a custom type.

Reply:
That's great to hear! My concern, however, is that, as far as I know, using `alloc::string::String` is discouraged to avoid bloating the runtime size.